grr/server/grr_response_server/flows/general/osquery_test.py

```diff
@@ -266,6 +271,130 @@ def testTotalRowCountIncludesAllChunks(self):
+          results = self._RunFlow("Doesn't matter", ["collect_collumn"])
+
+    self.assertLen(results, 2)
+    self.assertEqual(type(results[0]), rdf_osquery.OsqueryResult)
+    self.assertEqual(type(results[1]), rdf_client_fs.StatEntry)
+
+    client_path = db.ClientPath.FromPathSpec(
+        self.client_id, results[1].pathspec)
+    byte_content = file_store.OpenFile(client_path).read()
+    self.assertEqual(len(byte_content), less_than_necessary_bytes)
+
+  def testFlowDoesntCollectSingleFileAboveTotalLimit(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      target_bytes = 2**20  # 1 MiB
+
+      phrase = "This is a > 1 MiB file\n"
+      phrase_bytes = len(phrase.encode("utf-8"))
+      repeat_times = target_bytes // phrase_bytes + 1
+      to_write = phrase.encode("utf-8") * repeat_times
+
+      less_than_necessary_bytes = target_bytes // 2
+
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write(to_write)
+
+      table = f"""
+      [
+        {{ "collect_collumn": "{temp_file_path}" }}
+      ]
+      """
+
+      with mock.patch.object(
+          osquery_flow,
+          "FILE_COLLECT_LIMIT_MAX_TOTAL_BYTES",
+          less_than_necessary_bytes):
+        with osquery_test_lib.FakeOsqueryiOutput(stdout=table, stderr=""):
+          results = self._RunFlow("Doesn't matter", ["collect_collumn"])
+
+    # Assert that only the OsqueryResult is returned (no file results)
+    self.assertLen(results, 1)
+    self.assertEqual(type(results[0]), rdf_osquery.OsqueryResult)
+
+  def testFlowCollectFile(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write("Just sample text to put in the file.".encode('utf-8'))
+
+      table = f"""
+      [
+        {{ "collect_column": "{temp_file_path}" }}
+      ]
+      """
+
+      with osquery_test_lib.FakeOsqueryiOutput(stdout=table, stderr=""):
+          results = self._RunFlow("Doesn't matter", ["collect_column"])
+      
+    self.assertLen(results, 2)
+    self.assertEqual(type(results[0]), rdf_osquery.OsqueryResult)
+    self.assertEqual(type(results[1]), rdf_client_fs.StatEntry)
+
+    pathspec = results[1].pathspec
+    client_path = db.ClientPath.FromPathSpec(self.client_id, pathspec)
+    fd_rel_db = file_store.OpenFile(client_path)
+    file_text = fd_rel_db.read().decode("utf-8")
+    self.assertEqual(file_text, "Just sample text to put in the file.")
+
+  def testFlowDoesntCollectWhenColumnsAboveLimit(self):
+    with temp.AutoTempFilePath() as temp_file_path:
+      with io.open(temp_file_path, "wb") as fd:
+        fd.write("Just sample text to put in the file.".encode('utf-8'))
+
+      table = f"""
+      [
+        {{ "collect1": "{temp_file_path}", "collect2": "{temp_file_path}"}}
+      ]
+      """
+
+      with mock.patch.object(osquery_flow, "FILE_COLLECT_LIMIT_MAX_COLUMNS", 1):
+        with osquery_test_lib.FakeOsqueryiOutput(stdout=table, stderr=""):
```

Comment:
The flow should fail on startup. I'd change the flow logic and would just use "StartFlow" here and make sure that it raises.

Reply:
Given that the check for FILE_COLLECTION_MAX_COLUMNS happens in the Start method, do you really need to call FakeOsqueryiOutput above? I suppose the flow would never get to actually running anything osquery-related, no?

Reply:
Hm this comment is on the previous version of the code; I refactored it exactly as you advise:

https://github.com/google/grr/pull/901/files#diff-84b11633c72d1d91291437e8756c6edec25eb0172500d4a5265c47b5368c3c68R350

Reply:
Ah, GitHub can be so confusing at times... Thanks for clarifying. Resolved!